client/src/templates/Challenges/classic/editor.tsx

```diff
@@ -237,6 +237,11 @@ const Editor = (props: EditorProps): JSX.Element => {
   const monacoRef: MutableRefObject<typeof monacoEditor | null> =
     useRef<typeof monacoEditor>(null);
   const dataRef = useRef<EditorProperties>({ ...initialData });
+
+  const submitChallengeDebounceRef = useRef(
+    debounce(props.submitChallenge, 1000, { leading: true, trailing: true })
```

Comment:
```suggestion
    debounce(props.submitChallenge, 1000, { leading: true, trailing: false })
```
Sorry Ahmad, I know I suggested this, but it doesn't work as I hoped.  If you click rapidly, you'll get both the leading and the trailing calls, so you'll submitChallenge twice and end up skipping a step.